agent/tcs/client/client.go

```diff
@@ -137,33 +137,40 @@ func (cs *clientServer) publishMetrics() {
 	// Publish metrics immediately after we connect and wait for ticks. This makes
 	// sure that there is no data loss when a scheduled metrics publishing fails
 	// due to a connection reset.
-	cs.publishMetricsOnce()
+	err := cs.publishMetricsOnce()
+	if err != nil && err != stats.EmptyMetricsError {
+		seelog.Warnf("Error getting instance metrics: %v", err)
+	}
 	// don't simply range over the ticker since its channel doesn't ever get closed
 	for {
 		select {
 		case <-cs.publishTicker.C:
-			cs.publishMetricsOnce()
+			err := cs.publishMetricsOnce()
+			if err != nil {
+				seelog.Warnf("Error getting instance metrics: %v", err)
+			}
 		case <-cs.endPublish:
 			return
 		}
 	}
 }
 
 // publishMetricsOnce is invoked by the ticker to periodically publish metrics to backend.
-func (cs *clientServer) publishMetricsOnce() {
+func (cs *clientServer) publishMetricsOnce() error {
```

Comment:
Also, the [functional tests for windows](https://github.com/liwenwu-amazon/amazon-ecs-agent/commits/issue506) seem to have just stuck. Can you address that and make sure those pass as well?